src/reaction/views.py

```diff
@@ -36,3 +36,21 @@ def post(self, request):
         reaction.save()
         serializer = self.serializer_class(reaction)
         return success_response(serializer.data)
+
+
+class ReactionDelete(generics.GenericAPIView):
+
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def post(self, request, pk):
+        """
+        Delete a reaction
+        """
+        profile = Profile.objects.get(user=request.user)
+        if not Reaction.objects.filter(id=pk).exists():
+            return failure_response(f"Reaction with id {pk} does not exist!")
+        reaction = Reaction.objects.get(id=pk)
+        if not reaction.author == profile:
```

Comment:
`if reaction.author != profile`

Reply:
alternatively, if you don't wanna do `request.user.profile` above, you can just check `reaction.author.user != request.user`